### ISSUE TITLE
Custom Formats Not Included in Cutoff Unmet

### ISSUE THREAD

**Issue by sarakha63** [Sonarr/Sonarr#5304]
### Is there an existing issue for this?

- [X] I have searched the existing open and closed issues

### Current Behavior

I agree for the change in language handling  in V4
But there is a loss on a capability

Now language is detected and applied with the metadata info (this is great).  For giving some priority on download step we have now to make custom formats and use score for that.... that's ok

But let say i have a profile I'm interested in 1080 and I'm french. So i accept English subtitled... But i want it to be replaced by French

What is happening is that 

If the episode is downloaded in english but already has the maximum quality   the "Upgrade Until Custom Format Score" is simply ignored.

We can see that because those episodes are not in the Cutoff Unmet List

Before we were able to upgrade a language. 

I think everything is doable now except that Upgrade quality and upgrade Custom format should be considered both.. And upgrade custom format seems ignored when quality is already the highest... 

### Expected Behavior

Upgrad Custom Format until should be considered even if quality is already the highest. 

And those episodes should be seen on cutoff unmet list

As this is the only way to retreive the old mechanic 

to allow two languages and use a prefer language for replacement

I think allowing to upgrade custom formats despite the quality.  Or choose a cutsom format rule that is and always upgrade if the rule was not OK and a release makes it ok despite other parameters could be an idea

And also if the language is the highest we should never upgrade quality if it's reverting language.

That is a quite loss that sure can be handled with the new method but with some more fine tuning in upgrading.

Language upgrading is more important as quality

### Steps To Reproduce

1. Whatever Environnement

Edit a profile with upgrade quality

And a custom format with VO score 0 and VF score 1000

And use upgrade custom format until min 1000

You will see that if the episode is downloaded in VO (let's say english) but is already the highest quality, this episode wont show in cutoff unmett and will never be upgraded to VF (french)

Which is the behavior we could have before

### Environment

```markdown
- OS: Debian 
- Sonarr:4.0.0.287
- Docker Install: Yes
- Using Reverse Proxy:No
- Browser: Not Ui related
```

### What branch are you running?

Develop

### Trace Logs?

Not revelant

### Anything else?

![image](https://user-images.githubusercontent.com/2008029/207881246-94127b3a-7805-439a-9359-9b0c49d7c363.png)

**Comment by markus101** [comment]
> If the episode is downloaded in english but already has the maximum quality the "Upgrade Until Custom Format Score" is simply ignored.
> We can see that because those episodes are not in the Cutoff Unmet List

This is not true. Custom Format upgrades are not shown in Cutoff Unmet because they're calculated dynamically based on information in the title/size/series and not a single value. If any part of the Custom Format pipeline changes (CF's themselves or cutoff in quality profile) then everything needs to be recalculated.

We'll have to discuss whether Custom Format upgrades will be part of Cutoff Unmet in the future given the difficulty of including it and keeping it accurate.

**Comment by sarakha63** [comment]
@markus101 The part about language is still true i think.

Because custom formats allows an improved score only for better quality or same quality.

That means we have no way of making language upgrade a priority anymore if highest quality is already established.

I think this is the most important "issue".  

Dont you think ?

**Comment by markus101** [comment]
You can already do that, group all the qualities you want into one group and use Custom Formats to handle upgrading. You'd have the same cutoff unmet doesn't show "upgrades", but you'd have full control.

**Comment by sarakha63** [comment]
Ok i understand. (Missing language in cutoff unmet isnt a Big deal i think but Can be considered more of a feature request)
But doing the way you describe, i will lose the ability to have my release upgraded in quality until i have the language.

And once i will have the language (french) there wont be upgrade anymore on the quality inside that same group

If i'm not wrong ?

@markus101

**Comment by markus101** [comment]
No, you'd use Custom Formats to upgrade.
It'd entirely depend on your `Upgrade Until Custom Format Score` and the values you assign to the quality upgrades. If you want language over quality you'd give quality small increases in score and language larger increases so once it hits the language it won't upgrade the quality.

If you have more questions about how to do that please use one of the support channels (Discord would probably be best) as this isn't the place to get into how to do it.

**Comment by sarakha63** [comment]
Ok so your proposal is to totaly replace the native quality upgrade by grouping all in one

And define all the rules in custom formats. I get it.


So are WE suppose to consider that this is the way to go since 4.0 for cumulative language /quality upgraded ? There isnt going to be any change or improvements on that ?

**Comment by markus101** [comment]
Sonarr has never allowed downgrading quality in order to get a more preferred language, v3 worked the same way.

**Comment by sarakha63** [comment]
Ok my Bad.

I really tought before that language was priority number one over quality.


I will then use your proposal to handle that.

Lets leave the issue for the cutoff unmet thing until you discuss that.

Thanks again

**Comment by markus101** [comment]
This isn't something we plan to support at this time given the complications of maintaining accurate results.
